node/src/lib.rs

```diff
@@ -4,21 +4,197 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use dusk_consensus::user::provisioners::Provisioners;
-#[derive(Debug)]
-pub enum Error {}
+#![allow(unused)]
 
-/// Reads config parameters
-pub fn read_config() -> Result<(), Error> {
-    Ok(())
+pub mod chain;
+mod data;
+pub mod database;
+pub mod mempool;
+pub mod network;
+mod utils;
+
+use crate::utils::PendingQueue;
+use async_trait::async_trait;
+use data::Topics;
+use std::sync::Arc;
+use tokio::{
+    signal::unix::{signal, SignalKind},
+    sync::RwLock,
+    task::JoinSet,
+};
+use tracing::{error, info, Instrument};
+
+/// Filter is used by Network implementor to filter messages before re-routing
+/// them. It's like the middleware in HTTP pipeline.
+///
+/// To avoid delaying other messages handling, the execution of any filter
+/// should be fast as it is performed in the message handler .
+///
+/// e.g a message could be dropped by Reputation System Filter due to banned
```

Comment:
I like the use case, just a remind that this feature is already handled in kadcast

Reply:
Indeed.